packages/api/src/beacon/routes/validator.ts

```diff
@@ -846,33 +851,92 @@ export function getDefinitions(_config: ChainForkConfig): RouteDefinitions<Endpo
       url: "/eth/v1/validator/aggregate_attestation",
       method: "GET",
       req: {
-        writeReq: ({attestationDataRoot, slot}) => ({
-          query: {attestation_data_root: toHexString(attestationDataRoot), slot},
+        writeReq: ({attestationDataRoot, slot, committeeIndex}) => ({
+          query: {attestation_data_root: toHexString(attestationDataRoot), slot, committeeIndex},
+        }),
+        parseReq: ({query}) => ({
+          attestationDataRoot: fromHexString(query.attestation_data_root),
+          slot: query.slot,
+          committeeIndex: query.slot,
         }),
-        parseReq: ({query}) => ({attestationDataRoot: fromHexString(query.attestation_data_root), slot: query.slot}),
         schema: {
-          query: {attestation_data_root: Schema.StringRequired, slot: Schema.UintRequired},
+          query: {
+            attestation_data_root: Schema.StringRequired,
+            slot: Schema.UintRequired,
+            committeeIndex: Schema.UintRequired,
+          },
         },
       },
       resp: {
-        data: ssz.phase0.Attestation,
-        meta: EmptyMetaCodec,
+        data: WithVersion((fork) =>
+          ForkSeq[fork] >= ForkSeq.electra ? ssz.electra.Attestation : ssz.phase0.Attestation
+        ),
+        meta: VersionCodec,
       },
     },
     publishAggregateAndProofs: {
       url: "/eth/v1/validator/aggregate_and_proofs",
       method: "POST",
       req: {
-        writeReqJson: ({signedAggregateAndProofs}) => ({
-          body: SignedAggregateAndProofListType.toJson(signedAggregateAndProofs),
-        }),
-        parseReqJson: ({body}) => ({signedAggregateAndProofs: SignedAggregateAndProofListType.fromJson(body)}),
-        writeReqSsz: ({signedAggregateAndProofs}) => ({
-          body: SignedAggregateAndProofListType.serialize(signedAggregateAndProofs),
-        }),
-        parseReqSsz: ({body}) => ({signedAggregateAndProofs: SignedAggregateAndProofListType.deserialize(body)}),
+        writeReqJson: ({signedAggregateAndProofs}) => {
+          const fork = config.getForkName(signedAggregateAndProofs[0].message.aggregate.data.slot);
```

Comment:
technically this is unsafe access of `message` property, although pratically we only call this api if there is at least one `signedAggregateAndProof`

Reply:
yes if there is no data api will fail, but we can safe access it with `?` and default  it to 0(`??0`) as done in another section

Reply:
yeah, using `?` is the way to go, this should be enforced by typescript anyways since you didn't check that the element exists at index 0 but that's a known limitation.

Default to 0 makes sense as well as we don't really care what fork we pass in that case, alternative would be to see if slot exists, and default to a fork name